### PR TITLE
Pikus denis homework java

### PR DESCRIPTION
**Pull request by Denis-Pikus**
homework done

### CODE REVIEW

File: homework/java/pikus_denis/lecture1/task1/pom.xml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                        <id>jacoco-check</id>
+                        <goals>
+                            <goal>check</goal>
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <rule>
+                                    <limits>
+                                        <limit>
+                                            <minimum>0.0</minimum>
```
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/pom.xml:65]: use 0.6

File: homework/java/pikus_denis/lecture1/task1/pom.xml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>14</source>
```
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/pom.xml:33]: :0

File: homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/MathOperationsDemo.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package by.jrr.homework1;
```
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/MathOperationsDemo.java:1]: dont use by.jrr

File: homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/MathOperationsDemo.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+import by.jrr.homework1.operations.Calculator;
+
+public class MathOperationsDemo {
+    public static void main(String[] args) {
+        Calculator calc = new Calculator();
+        System.out.println(calc.addition(5, 6, 1));
```
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/MathOperationsDemo.java:8]: use log4j

File: homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/MathOperationsDemo.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+        System.out.println(calc.subtraction(3, 1, 10));
+        System.out.println(calc.multiplication(-2, 3, 6));
+        try {
+            System.out.println(calc.devision(10, 5, 55));
+        } catch (Exception e) {
```
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/MathOperationsDemo.java:13]: use IllegualArgumentException

File: homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/RandomGeneratorDemo.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+import by.jrr.homework1.operations.Calculator;
+import by.jrr.homework1.operations.GeneratorRandomNumber;
+
+import java.util.Arrays;
+
+public class RandomGeneratorDemo {
+    public static void main(String[] args) {
+        Calculator calc = new Calculator();
+        GeneratorRandomNumber gen = new GeneratorRandomNumber();
```
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/RandomGeneratorDemo.java:11]: use var :)

File: homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/MathOperationsDemo.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package by.jrr.homework1;
+
+import by.jrr.homework1.operations.Calculator;
+
+public class MathOperationsDemo {
+    public static void main(String[] args) {
+        Calculator calc = new Calculator();
+        System.out.println(calc.addition(5, 6, 1));
+        System.out.println(calc.subtraction(3, 1, 10));
+        System.out.println(calc.multiplication(-2, 3, 6));
```
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/MathOperationsDemo.java:10]: add double and test case with double format

File: homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/GeneratorRandomNumber.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
+package by.jrr.homework1.operations;
+
+import java.util.Random;
+
+public class GeneratorRandomNumber {
+    Random rnd;
+    double [] arr;
```
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/GeneratorRandomNumber.java:7]: use more self-explainarry variables names

File: homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/GeneratorRandomNumber.java [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    public double [] generatorOfRandomNumbers(int sizeOfArray, int rangeOfGenerators){
```
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/GeneratorRandomNumber.java:8]: attach [] to double

File: homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/RandomGeneratorDemo.java [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+        double[] arr = gen.generatorOfRandomNumbers(3, 10);
+        System.out.println(Arrays.toString(arr));
+        System.out.println(calc.addition(arr));
+        System.out.println(calc.subtraction(arr));
+        System.out.println(calc.multiplication(arr));
```
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/RandomGeneratorDemo.java:16]: use logs

File: homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/Calculator.java [hunk 11]
```diff
@@ -0,0 +1,44 @@
+package by.jrr.homework1.operations;
+
+public class Calculator {
+    double sum;
+    double sub;
+    double mult = 1;
+    double dev;
```
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/Calculator.java:7]: private modificator. why you set 1?

create constructor when you set this

File: homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/Calculator.java [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+
+    public double addition(double ... numbers){
+        for (Double num: numbers) {
```
Comment (javaGuruBY) [inline homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/Calculator.java:10]: `(Double num : numbers) `

surround ":" with spaces

File: homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/Calculator.java [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+    public double subtraction(double ... numbers){
+        sub = numbers[0];
+        for (int i = 1; i < numbers.length; i++) {
+            sub -= numbers[i];
+        }
+        return sub;
+    }
+
+    public double multiplication(double ... numbers){
+
```
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/Calculator.java:25]: test this momet with double. null and 0 and 0

File: homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/Calculator.java [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+        for (Double num: numbers) {
+            mult *= num;
+        }
+        return mult;
+    }
+
+    public double devision(double ... numbers) throws Exception {
+        dev = numbers[0];
```
Comment (javaGuruBY) [inline homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/Calculator.java:33]: add check that numbers is not null

File: homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/Calculator.java [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+        for (int i = 1; i < numbers.length; i++) {
+            if (numbers[i] == 0) {
+                throw new Exception("Division by 0 is impossible!");
+            }
+            else {
+                dev /= numbers[i];
+            }
```
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/Calculator.java:40]: add test for numbers == null and remove dev from method. use numbers[...]

File: homework/java/pikus_denis/lecture1/task1/src/test/java/by/jrr/homework1/operations/CalculatorTest.java [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+
+import org.junit.Assert;
+import org.junit.Test;
+import java.util.stream.DoubleStream;
+
+public class CalculatorTest {
+    Calculator calc = new Calculator();
+    @Test
+    public void addition() {
+        double [] numbers = new double [] {1, 10, 5};
```
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/src/test/java/by/jrr/homework1/operations/CalculatorTest.java:11]: attach [] to double

File: homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/Calculator.java (hunk 11 above)
Comment (javaGuruBY) [inline homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/Calculator.java:7]: These are redundant.

File: homework/java/pikus_denis/lecture1/task1/src/test/java/by/jrr/homework1/operations/CalculatorTest.java [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+        double expected = DoubleStream.of(numbers).sum();
```
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/src/test/java/by/jrr/homework1/operations/CalculatorTest.java:12]: nice)

File: homework/java/pikus_denis/lecture1/task1/src/test/java/by/jrr/homework1/operations/CalculatorTest.java [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+        double actual = calc.addition(numbers);
+        Assert.assertEquals(expected, actual, 0.01);
+    }
+
+    @Test
+    public void subtraction() {
+        double [] numbers = new double [] {1, 10, 5};
```
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/src/test/java/by/jrr/homework1/operations/CalculatorTest.java:19]: remove spaces

File: homework/java/pikus_denis/lecture1/task1/src/test/java/by/jrr/homework1/operations/CalculatorTest.java [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+        actual = calc.multiplication(numbers);
+        Assert.assertEquals(expected, actual, 0.01);
+    }
+
+    @Test
+    public void devision() throws Exception {
+        double [] numbers = new double [] {10, 5};
+        double expected = 2.0;
+        double actual = calc.devision(numbers);
+        Assert.assertEquals(expected, actual, 0.01);
```
Comment (javaGuruBY) [inline homework/java/pikus_denis/lecture1/task1/src/test/java/by/jrr/homework1/operations/CalculatorTest.java:40]: Use static imports for Assert

File: homework/java/pikus_denis/lecture1/task1/src/test/java/by/jrr/homework1/operations/CalculatorTest.java (hunk 18 above)
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/src/test/java/by/jrr/homework1/operations/CalculatorTest.java:19]: use double numbers for test

File: homework/java/pikus_denis/lecture1/task1/task1.iml [hunk 20]
```diff
@@ -0,0 +1,22 @@
...
+    <orderEntry type="sourceFolder" forTests="false" />
+    <orderEntry type="library" scope="TEST" name="Maven: junit:junit:4.12" level="project" />
+    <orderEntry type="library" scope="TEST" name="Maven: org.hamcrest:hamcrest-core:1.3" level="project" />
+  </component>
+  <component name="POM File Configuration" pomFile="" />
+  <component name="sonarModuleSettings">
+    <option name="localAnalysisScripName" value="&lt;PROJECT&gt;" />
+    <option name="serverName" value="&lt;PROJECT&gt;" />
+  </component>
+</module>
```
Comment (javaGuruBY) [inline homework/java/pikus_denis/lecture1/task1/task1.iml:22]: remove from git

File: homework/java/pikus_denis/lecture1/task1/src/test/java/by/jrr/homework1/operations/GeneratorRandomNumberTest.java [hunk 21]
```diff
@@ -0,0 +1,16 @@
+package by.jrr.homework1.operations;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class GeneratorRandomNumberTest {
+
+    @Test
+    public void generatorOfRandomNumbers() {
+        GeneratorRandomNumber gen = new GeneratorRandomNumber();
```
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/src/test/java/by/jrr/homework1/operations/GeneratorRandomNumberTest.java:10]: remove gen
use generatorRandomNumber and use var

File: homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/GeneratorRandomNumber.java [hunk 22]
```diff
@@ -0,0 +1,16 @@
...
+        rnd = new Random();
+        arr = new double[sizeOfArray];
+        for (int i = 0; i < arr.length; i++) {
+            arr[i] = rnd.nextInt(rangeOfGenerators);
+        }
+        return arr;
+    }
```
Comment (javaGuruBY) [inline homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/GeneratorRandomNumber.java:15]: use ThreadLocal.random()

File: homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/GeneratorRandomNumber.java [hunk 23]
```diff
@@ -0,0 +1,16 @@
+package by.jrr.homework1.operations;
+
+import java.util.Random;
+
+public class GeneratorRandomNumber {
+    Random rnd;
+    double [] arr;
+    public double [] generatorOfRandomNumbers(int sizeOfArray, int rangeOfGenerators){
+        rnd = new Random();
```
Comment (integratorBY) [inline homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/GeneratorRandomNumber.java:9]: use ThreadLocalRandomNext

Comment (integratorBY) [reply homework/java/pikus_denis/lecture1/task1/src/main/java/by/jrr/homework1/operations/GeneratorRandomNumber.java:9]: https://www.baeldung.com/java-thread-local-random

**Review by javaGuruBY — APPROVED** [review]
lgfm, minor comments

**Review by integratorBY — APPROVED** [review]
lgfm

**Review by kashlyak — COMMENTED** [review]
lgfm

**Review by Svetlana101010 — APPROVED** [review]
lgfm

**Review by Alex99905 — APPROVED** [review]
